lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -333,6 +334,10 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
         if (this.isPropertyChanged(changes['sorting'])) {
             this.setTableSorting(changes['sorting'].currentValue);
         }
+
+        if (multiselectChanges?.currentValue === false) {
```

Comment:
I also think that a refactor to improve when the reset is needed it would be nice. 

We should make the code more reactive rather than sequential,         It is very difficult to read

```
                if (dataChanges) {
                    this.data = changes['data'].currentValue;
                    this.resetSelection();
                }

                if (rowChanges) {
                    this.setTableRows(changes['rows'].currentValue);
                    this.setTableSorting(this.sorting);
                }

                if (columnChanges) {
                    this.setTableColumns(changes['columns'].currentValue);
                }
            }

            return;
        }

        if (changes.selectionMode && !changes.selectionMode.isFirstChange()) {
            this.resetSelection();
            this.emitRowSelectionEvent('row-unselect', null);
        }

        if (this.isPropertyChanged(changes['sorting'])) {
            this.setTableSorting(changes['sorting'].currentValue);
        }

        if (multiselectChanges?.currentValue === false) {
            this.resetSelection();
        }